Enterprise-CMCS/macpro-mdct-mfp
# Refactor AddNewWorkPlanModal into AddEditReportModal

### Description

Compare to PR #616 

Refactor the two dashboard modals into one that accommodates all scenarios for WP and SAR creation.
A lot of the new logic in `AddEditReportModal.tsx` and the tests are from the deleted modal files.

`components/modals/AddEditReportModal.tsx`
- update double equals to triple throughout
- add ability to show both WP form types
- update `prepareWpPayload` and `writeReport` with logic from the deleted AddNewWorkPlanModal
- update jsx to simplify logic and use full capability of modal
- remove unused styling (sx)

`components/modals/AddEditReportModal.test.tsx`
- add test component for new wp
- add test component for editing sar
- copy tests for new work plans from `AddNewWorkPlanModal.test.tsx` into this file
- update SAR tests
- add a11y tests for all view types

`components/pages/Dashboard/DashboardPage.tsx`
- remove logic that split which modal is opened

deleted `components/modals/AddNewWorkPlanModal.test.tsx`
deleted `components/modals/AddNewWorkPlanModal.tsx`
removed from `components/index.ts`

`tests/cypress/support/commands.js`
- Add extra wait after clearing sessions to ensure we start fresh

Open to feedback! Do we like this better?
In doing this refactor I realized that maybe grouping both WP types into one component, and keeping SAR separate, could be another way to organize it. I know that would give us two components again, but maybe the logic would be simpler when drawn along those lines. What do you think? Do you think this component is logical as is?


### Related ticket(s)


CMDCT-3473

---
### How to test

Deployed url: https://d176fbxzy8k4fx.cloudfront.net/
Fastest to test locally with the postman collection:
- Log in as a state user
- Create a WP and ensure you get the first work plan form (asks for year and period)
- Use the postman collection to fill out, submit, and approve the report
- In the UI, create a copy WP and ensure the correct modal shows
- Go to SAR dashboard and ensure that modal works as expected

---
### Pre-review checklist
